management/src/test/java/org/ehcache/management/providers/statistics/StatsUtil.java

```diff
@@ -68,4 +76,64 @@ public static Context createContext(ManagementRegistryService managementRegistry
         .with(cacheManagerCtx.getName(), cacheManagerCtx.getValue())
         .with(firstCacheCtx.getName(), firstCacheCtx.getValue());
   }
+
+  public static long getExpectedValueFromCounterHistory(String statName, Context context, ManagementRegistryService managementRegistry, long expectedResult) {
+
+    StatisticQuery query = managementRegistry.withCapability("StatisticsCapability")
+          .queryStatistics(Arrays.asList(statName))
+          .on(context)
+          .build();
+
+    long value = 0;
+    do {
+      ResultSet<ContextualStatistics> counters = query.execute();
+
+      ContextualStatistics statisticsContext = counters.getResult(context);
+
+      Assert.assertThat(counters.size(), Matchers.is(1));
+
+      CounterHistory counterHistory = statisticsContext.getStatistic(CounterHistory.class, statName);
+
+      if (counterHistory.getValue().length > 0) {
+        int mostRecentIndex = counterHistory.getValue().length - 1;
+        value = counterHistory.getValue()[mostRecentIndex].getValue();
+        System.out.println("statName: " + statName + " value: " + value + " expectedResult: " + expectedResult);
+      }
+
+    }while(value != expectedResult);
+
+    Assert.assertThat(value, Matchers.is(expectedResult));
+
+    return value;
+  }
+
+  public static double getExpectedValueFromRatioHistory(String statName, Context context, ManagementRegistryService managementRegistry, double expectedResult) {
+
+    StatisticQuery query = managementRegistry.withCapability("StatisticsCapability")
+          .queryStatistics(Arrays.asList(statName))
+          .on(context)
+          .build();
+
+    double value = 0;
+    do {
+      ResultSet<ContextualStatistics> counters = query.execute();
+
+      ContextualStatistics statisticsContext = counters.getResult(context);
+
+      Assert.assertThat(counters.size(), Matchers.is(1));
+
+      RatioHistory ratioHistory = statisticsContext.getStatistic(RatioHistory.class, statName);
+
+      if (ratioHistory.getValue().length > 0) {
+        int mostRecentIndex = ratioHistory.getValue().length - 1;
+        value = ratioHistory.getValue()[mostRecentIndex].getValue();
+        System.out.println("statName: " + statName + " value: " + value + " expectedResult: " + expectedResult);
+      }
+
+    }while(value != expectedResult);
```

Comment:
same remark:
If the expected value is not in the list of samples, at the latest position, this could lead to an infinite loop ?
should you instead do a sort of for loop to iterate over the sample values from the end to the start ?